src/contexts/Accounts/Ledgers/index.tsx

```diff
@@ -0,0 +1,194 @@
+import { AnyApi, MaybeAccount } from 'types';
+import { rmCommas, setStateWithRef } from 'Utils';
+import * as defaults from './defaults';
+import { Ledger, LedgersContextInterface } from './types';
+
+export const LedgersContext = React.createContext<LedgersContextInterface>(
+  defaults.defaultLedgersContext
+);
+
+export const useLedgers = () => React.useContext(LedgersContext);
+
+export const LedgersProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const { api, isReady, network } = useApi();
+  const { accounts, addExternalAccount } = useConnect();
+
+  // account ledgers to separate storage
+  const [ledgers, setLedgers] = useState<Array<Ledger>>([]);
+  const ledgersRef = useRef(ledgers);
+
+  // ledger subscriptions state
+  const [unsubsLedgers, setUnsubsLedgers] = useState<AnyApi>([]);
+  const unsubsLedgersRef = useRef<AnyApi>(unsubsLedgers);
+
+  // fetch account balances & ledgers. Remove or add subscriptions
+  useEffect(() => {
+    if (isReady) {
+      // local updated values
+      let newLedgers = ledgersRef.current;
+      const newUnsubsLedgers = unsubsLedgersRef.current;
+
+      // get accounts removed: use these to unsubscribe
+      const accountsRemoved = ledgersRef.current.filter(
+        (a: Ledger) =>
+          !accounts.find((c: ImportedAccount) => c.address === a.address)
+      );
+      // get accounts added: use these to subscribe
+      const accountsAdded = accounts.filter(
+        (c: ImportedAccount) =>
+          !ledgersRef.current.find((a: Ledger) => a.address === c.address)
+      );
+
+      // update ledgers state for removal
+      newLedgers = ledgersRef.current.filter((l: Ledger) =>
+        accounts.find((c: ImportedAccount) => c.address === l.address)
+      );
+
+      // update ledgers state and unsubscribe if accounts have been removed
+      if (newLedgers.length < ledgersRef.current.length) {
+        // unsubscribe from removed ledgers if it exists
+        accountsRemoved.forEach((a: Ledger) => {
+          const unsub = unsubsLedgersRef.current.find(
+            (u: AnyApi) => u.key === a.address
+          );
+          if (unsub) {
+            unsub.unsub();
+            // remove unsub from balances
+            newUnsubsLedgers.filter((u: AnyApi) => u.key !== a.address);
+          }
+        });
+        // commit state updates
+        setStateWithRef(newUnsubsLedgers, setUnsubsLedgers, unsubsLedgersRef);
+        setStateWithRef(newLedgers, setLedgers, ledgersRef);
+      }
+
+      // if accounts have changed, update state with new unsubs / accounts
+      if (accountsAdded.length) {
+        // subscribe to added accounts
+        accountsAdded.map((a: ImportedAccount) => subscribeToLedger(a.address));
+      }
+    }
+  }, [accounts, network, isReady]);
+
+  // unsubscribe from ledger subscriptions on unmount
+  useEffect(() => {
+    Object.values(unsubsLedgersRef.current).forEach(({ unsub }: AnyApi) => {
```

Comment:
`unsubsLedgersRef` can simply called `unsubsRef`. The same goes to `Balances` too.